src/braket/circuits/result_types.py

```diff
@@ -174,7 +182,8 @@ def target(self, target: QubitSetInput) -> None:
 
     def to_ir(self) -> ir.Probability:
         if self.target:
-            return ir.Probability.construct(targets=list(self.target))
+            # convert qubits to int as required by the ir type
+            return ir.Probability.construct(targets=[int(qubit) for qubit in self.target])
```

Comment:
This breaks with `Qubit`? Surprising consider `Qubit` implements `int`

Reply:
Indeed. Since the result_type hash uses __repr__, '1' != 'Qubit(1)'. It's possible to move the casting into the hash function if you think that would be cleaner, but since ir typically has raw ints, I feel it could make sense for the to_ir function to cast to int

Reply:
Interesting, what I'm seeing is
```
In [4]: 1 == Qubit(1)                                                                   
Out[4]: True
```

Reply:
I use the repr function in the hash

```
In: repr(Qubit(1)) == repr(1)
Out: False
```